node/src/main/scala/com/wavesplatform/network/package.scala

```diff
@@ -20,17 +17,21 @@ import monix.execution.Scheduler
 import monix.reactive.Observable
 import org.slf4j.LoggerFactory
 
+import java.net.{InetAddress, InetSocketAddress, SocketAddress, URI}
+import java.util.concurrent.Callable
 import scala.concurrent.duration.*
 
 package object network {
   private val broadcastTimeStats = Kamon.timer("network-broadcast-time")
   private lazy val logger: Logger =
     Logger(LoggerFactory.getLogger(getClass.getName))
 
-  def inetSocketAddress(addr: String, defaultPort: Int): InetSocketAddress = {
-    val uri = new URI(s"node://$addr")
-    if (uri.getPort < 0) new InetSocketAddress(addr, defaultPort)
-    else new InetSocketAddress(uri.getHost, uri.getPort)
+  def inetSocketAddress(addr: String, defaultPort: Int): Seq[InetSocketAddress] = {
+    val uri        = new URI(s"node://$addr")
+    val actualPort = if (uri.getPort > 0) uri.getPort else defaultPort
+    InetAddress.getAllByName(uri.getHost).view.map { ia =>
+      new InetSocketAddress(ia, actualPort)
+    }.toSeq
```

Comment:
It can throw `java.net.UnknownHostException`, for example, if a known peer is no longer maintained.
```suggestion
    try {
      InetAddress.getAllByName(uri.getHost).view.map { ia =>
        new InetSocketAddress(ia, actualPort)
      }.toSeq
    } catch {
      case e: UnknownHostException =>
        logger.debug(e.getMessage)
        Nil
    }
```